File: dashboard-flutter/lib/ui/login/login_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../../auth/auth_provider.dart';
import '../../core/theme.dart';

class LoginScreen extends StatelessWidget {
  const LoginScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final auth = context.watch<AuthProvider>();

    Widget body;
    if (auth.isBlocked) {
      body = const _BlockedView();
    } else if (auth.isSignedIn && auth.needsOwnerCode) {
      body = const _OwnerCodeView();
    } else {
      body = const _SignInView();
    }

    return Scaffold(
      body: Center(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(24),
          child: ConstrainedBox(
            constraints: const BoxConstraints(maxWidth: 420),
            child: Card(
              child: Padding(
                padding: const EdgeInsets.all(28),
                child: Column(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    const Icon(Icons.auto_stories_rounded,
                        size: 52, color: NebrasColors.emerald),
                    const SizedBox(height: 12),
                    const Text('مرحبًا بك في لوحة تحكّم نبراس',
                        textAlign: TextAlign.center,
                        style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold)),
                    const SizedBox(height: 4),
                    const Text('سجّل الدخول بحساب Google لإدارة المنصّة.',
                        textAlign: TextAlign.center,
                        style: TextStyle(color: NebrasColors.textMuted)),
                    const SizedBox(height: 28),
                    body,
                    if (auth.errorMessage != null) ...[
                      const SizedBox(height: 16),
                      Text(auth.errorMessage!,
                          textAlign: TextAlign.center,
                          style: const TextStyle(color: NebrasColors.rose, fontSize: 13)),
                    ],
                  ],
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}

class _SignInView extends StatelessWidget {
  const _SignInView();

  @override
  Widget build(BuildContext context) {
    final auth = context.watch<AuthProvider>();
    return Column(
      children: [
        if (auth.isLoading)
          const Padding(
            padding: EdgeInsets.all(8),
            child: CircularProgressIndicator(),
          )
        else
          SizedBox(
            width: double.infinity,
            child: FilledButton.icon(
              onPressed: () => context.read<AuthProvider>().signInWithGoogle(),
              icon: const Icon(Icons.login),
              label: const Text('المتابعة عبر Google'),
            ),
          ),
        const SizedBox(height: 12),
        const Text(
          'نقرأ فقط بياناتك الأساسيّة من Google. لا نخزّن أيّ كلمة مرور هنا.',
          textAlign: TextAlign.center,
          style: TextStyle(color: NebrasColors.textMuted, fontSize: 12),
        ),
      ],
    );
  }
}

class _OwnerCodeView extends StatefulWidget {
  const _OwnerCodeView();

  @override
  State<_OwnerCodeView> createState() => _OwnerCodeViewState();
}

class _OwnerCodeViewState extends State<_OwnerCodeView> {
  final _codeCtrl = TextEditingController();
  bool _busy = false;
  String? _info;

  @override
  void dispose() {
    _codeCtrl.dispose();
    super.dispose();
  }

  Future<void> _request() async {
    setState(() => _busy = true);
    try {
      final msg = await context.read<AuthProvider>().requestOwnerCode();
      setState(() => _info = msg);
    } catch (e) {
      setState(() => _info = 'تعذّر إرسال الرمز: $e');
    } finally {
      if (mounted) setState(() => _busy = false);
    }
  }

  Future<void> _verify() async {
    if (_codeCtrl.text.trim().isEmpty) return;
    setState(() => _busy = true);
    try {
      await context.read<AuthProvider>().verifyOwnerCode(_codeCtrl.text);
    } finally {
      if (mounted) setState(() => _busy = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    final auth = context.watch<AuthProvider>();
    return Column(
      children: [
        Text('حسابك (${auth.user?.email ?? ''}) جديد ويحتاج اعتماد المالك.',
            textAlign: TextAlign.center,
            style: const TextStyle(color: NebrasColors.textMuted, fontSize: 13)),
        const SizedBox(height: 16),
        SizedBox(
          width: double.infinity,
          child: OutlinedButton.icon(
            onPressed: _busy ? null : _request,
            icon: const Icon(Icons.mark_email_read_outlined),
            label: const Text('إرسال رمز جديد'),
          ),
        ),
        const SizedBox(height: 16),
        TextField(
          controller: _codeCtrl,
          keyboardType: TextInputType.number,
          textAlign: TextAlign.center,
          decoration: const InputDecoration(
            labelText: 'رمز التحقّق',
          ),
        ),
        const SizedBox(height: 12),
        SizedBox(
          width: double.infinity,
          child: FilledButton(
            onPressed: _busy ? null : _verify,
            child: _busy
                ? const SizedBox(
                    height: 18, width: 18, child: CircularProgressIndicator(strokeWidth: 2))
                : const Text('تحقّق وتابع'),
          ),
        ),
        if (_info != null) ...[
          const SizedBox(height: 12),
          Text(_info!,
              textAlign: TextAlign.center,
              style: const TextStyle(color: NebrasColors.emerald, fontSize: 12)),
        ],
        const SizedBox(height: 8),
        TextButton(
          onPressed: () => context.read<AuthProvider>().signOut(),
          child: const Text('إلغاء وتسجيل الخروج'),
        ),
      ],
    );
  }
}

class _BlockedView extends StatelessWidget {
  const _BlockedView();

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        const Icon(Icons.block, color: NebrasColors.rose, size: 40),
        const SizedBox(height: 12),
        const Text('تم تعليق وصولك من قبل الإدارة.',
            textAlign: TextAlign.center,
            style: TextStyle(color: NebrasColors.rose)),
        const SizedBox(height: 16),
        TextButton(
          onPressed: () => context.read<AuthProvider>().signOut(),
          child: const Text('تسجيل الخروج'),
        ),
      ],
    );
  }
}

File: nebras-dashboard/app/src/main/kotlin/com/nebras/dashboard/ui/login/LoginScreen.kt
package com.nebras.dashboard.ui.login

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Login
import androidx.compose.material.icons.filled.AutoStories
import androidx.compose.material.icons.filled.Block
import androidx.compose.material.icons.filled.MarkEmailRead
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.nebras.dashboard.auth.AuthController
import com.nebras.dashboard.core.NebrasColors
import com.nebras.dashboard.ui.widgets.DashCard
import com.nebras.dashboard.ui.widgets.DashPrimaryButton
import com.nebras.dashboard.ui.widgets.DashSecondaryButton
import com.nebras.dashboard.ui.widgets.DashTextField
import kotlinx.coroutines.launch

/**
 * شاشة الدخول — نقل `ui/login/login_screen.dart`.
 *
 * ثلاث حالات بنفس ترتيب الأصل: محظور، أو داخل ويحتاج رمز اعتماد المالك،
 * أو غير مسجَّل.
 */
@Composable
fun LoginScreen(authController: AuthController) {
    val state by authController.state.collectAsStateWithLifecycle()

    Box(
        modifier = Modifier.fillMaxSize().verticalScroll(rememberScrollState()),
        contentAlignment = Alignment.Center,
    ) {
        DashCard(
            modifier = Modifier
                .widthIn(max = 420.dp)
                .padding(24.dp),
        ) {
            Column(
                modifier = Modifier.fillMaxWidth().padding(12.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                Icon(
                    Icons.Default.AutoStories,
                    contentDescription = null,
                    tint = NebrasColors.emerald,
                    modifier = Modifier.size(52.dp),
                )
                Spacer(Modifier.height(12.dp))
                Text(
                    text = "مرحبًا بك في لوحة تحكّم نبراس",
                    textAlign = TextAlign.Center,
                    style = MaterialTheme.typography.titleLarge.copy(
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold,
                    ),
                )
                Spacer(Modifier.height(4.dp))
                Text(
                    text = "سجّل الدخول بحساب Google لإدارة المنصّة.",
                    textAlign = TextAlign.Center,
                    color = NebrasColors.textMuted,
                    style = MaterialTheme.typography.bodyMedium,
                )
                Spacer(Modifier.height(28.dp))

                when {
                    state.isBlocked -> BlockedView(authController)
                    state.isSignedIn && state.needsOwnerCode ->
                        OwnerCodeView(authController, state.email)
                    else -> SignInView(authController, state.isLoading || state.isBusy)
                }

                state.errorMessage?.let { error ->
                    Spacer(Modifier.height(16.dp))
                    Text(
                        text = error,
                        textAlign = TextAlign.Center,
                        color = NebrasColors.rose,
                        style = MaterialTheme.typography.bodySmall.copy(fontSize = 13.sp),
                    )
                }
            }
        }
    }
}

@Composable
private fun SignInView(authController: AuthController, isLoading: Boolean) {
    val context = LocalContext.current
    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        if (isLoading) {
            Box(Modifier.padding(8.dp)) { CircularProgressIndicator() }
        } else {
            DashPrimaryButton(
                text = "المتابعة عبر Google",
                icon = Icons.AutoMirrored.Filled.Login,
                onClick = { authController.signInWithGoogle(context) },
                modifier = Modifier.fillMaxWidth(),
            )
        }
        Spacer(Modifier.height(12.dp))
        Text(
            text = "نقرأ فقط بياناتك الأساسيّة من Google. لا نخزّن أيّ كلمة مرور هنا.",
            textAlign = TextAlign.Center,
            color = NebrasColors.textMuted,
            style = MaterialTheme.typography.bodySmall.copy(fontSize = 12.sp),
        )
    }
}

/**
 * اعتماد المالك: الحساب الجديد يطلب رمزاً يصل إلى المالك ثمّ يتحقّق منه.
 */
@Composable
private fun OwnerCodeView(authController: AuthController, email: String) {
    val scope = rememberCoroutineScope()
    var code by remember { mutableStateOf("") }
    var busy by remember { mutableStateOf(false) }
    var info by remember { mutableStateOf<String?>(null) }

    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Text(
            text = "حسابك ($email) جديد ويحتاج اعتماد المالك.",
            textAlign = TextAlign.Center,
            color = NebrasColors.textMuted,
            style = MaterialTheme.typography.bodySmall.copy(fontSize = 13.sp),
        )
        Spacer(Modifier.height(16.dp))

        DashSecondaryButton(
            text = "إرسال رمز جديد",
            icon = Icons.Default.MarkEmailRead,
            enabled = !busy,
            onClick = {
                scope.launch {
                    busy = true
                    info = runCatching { authController.requestOwnerCode() }
                        .getOrElse { "تعذّر إرسال الرمز: ${it.message}" }
                    busy = false
                }
            },
            modifier = Modifier.fillMaxWidth(),
        )

        Spacer(Modifier.height(16.dp))

        DashTextField(
            value = code,
            onValueChange = { code = it },
            label = "رمز التحقّق",
            enabled = !busy,
        )

        Spacer(Modifier.height(12.dp))

        DashPrimaryButton(
            text = "تحقّق وتابع",
            loading = busy,
            enabled = code.trim().isNotEmpty(),
            onClick = {
                scope.launch {
                    busy = true
                    runCatching { authController.verifyOwnerCode(code) }
                    busy = false
                }
            },
            modifier = Modifier.fillMaxWidth(),
        )

        info?.let {
            Spacer(Modifier.height(12.dp))
            Text(
                text = it,
                textAlign = TextAlign.Center,
                color = NebrasColors.emerald,
                style = MaterialTheme.typography.bodySmall.copy(fontSize = 12.sp),
            )
        }

        Spacer(Modifier.height(8.dp))
        TextButton(onClick = authController::signOut) {
            Text("إلغاء وتسجيل الخروج")
        }
    }
}

@Composable
private fun BlockedView(authController: AuthController) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        Icon(
            Icons.Default.Block,
            contentDescription = null,
            tint = NebrasColors.rose,
            modifier = Modifier.size(40.dp),
        )
        Spacer(Modifier.height(12.dp))
        Text(
            text = "تم تعليق وصولك من قبل الإدارة.",
            textAlign = TextAlign.Center,
            color = NebrasColors.rose,
            style = MaterialTheme.typography.bodyMedium,
        )
        Spacer(Modifier.height(16.dp))
        TextButton(onClick = authController::signOut) {
            Text("تسجيل الخروج")
        }
    }
}
